src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/CustomFhirJsonSerializer.cs

```diff
@@ -0,0 +1,38 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Reflection.Metadata.Ecma335;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Serialization;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Features.Operations.Import;
+using Microsoft.Health.Fhir.Core.Features.Resources.Patch;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations
+{
+    public class CustomFhirJsonSerializer<T> : ICustomFhirJsonSerializer<T>
```

Comment:
Why is this class needed instead of using the existing resource serializer and deserializer?